src/Service/Modules/StockPoolModule.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace Linn.Stores2.Service.Modules
+{
+    using System.Threading.Tasks;
+
+    using Linn.Common.Service.Core;
+    using Linn.Common.Service.Core.Extensions;
+    using Linn.Stores2.Domain.LinnApps.Stock;
+    using Linn.Stores2.Facade.Common;
+    using Linn.Stores2.Resources;
+    using Linn.Stores2.Service.Extensions;
+
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Routing;
+
+    public class StockPoolModule : IModule
+    {
+        public void MapEndpoints(IEndpointRouteBuilder app)
+        {
+            app.MapGet("/stores2/stock-pools", this.GetStockPools);
+            app.MapGet("/stores2/stock-pools/{code}", this.GetStockPool);
+            app.MapPost("/stores2/stock-pools", this.CreateStockPool);
+            app.MapPut("/stores2/stock-pools/{code}", this.UpdateStockPool);
+        }
+
+        private async Task GetStockPools(
+            HttpRequest req,
+            HttpResponse res,
+            IAsyncFacadeService<StockPool, string, StockPoolResource, StockPoolUpdateResource, StockPoolResource> facadeService)
+        {
+            var results = await facadeService.GetAll(req.HttpContext.GetPrivileges());
+            await res.Negotiate(results);
+        }
+
+        private async Task GetStockPool(
+            HttpRequest req,
+            HttpResponse res,
+            string code,
+            IAsyncFacadeService<StockPool, string, StockPoolResource, StockPoolUpdateResource, StockPoolResource> facadeService)
+        {
+            var results = await facadeService.GetById(code, req.HttpContext.GetPrivileges());
+            await res.Negotiate(results);
+        }
+
+        private async Task CreateStockPool(
+            HttpRequest req,
+            HttpResponse res,
+            StockPoolResource resource,
+            IAsyncFacadeService<StockPool, string, StockPoolResource, StockPoolUpdateResource, StockPoolResource> facadeService)
+        {
+            await res.Negotiate(await facadeService.Add(resource));
```

Comment:
you are passing privileges for every other request, so might as well pass here to? If anything is going to need them it will likely by create/update